projects/plugins/boost/app/modules/Module.php

```diff
@@ -26,13 +29,31 @@ public function __construct( Pluggable $feature ) {
 	}
 
 	public function on_activate() {
+		if ( $this->feature instanceof Changes_Output_On_Activation ) {
+			$this->indicate_page_output_changed();
+		}
+
 		return $this->feature instanceof Has_Activate ? $this->feature::activate() : true;
 	}
 
 	public function on_deactivate() {
+		// If the module changes the page output, with or without preparation, deactivating the module should indicate a page output change.
+		if ( $this->feature instanceof Changes_Output_On_Activation || $this->feature instanceof Changes_Output_After_Activation ) {
+			$this->indicate_page_output_changed();
+		}
+
 		return $this->feature instanceof Has_Deactivate ? $this->feature::deactivate() : true;
 	}
 
+	private function indicate_page_output_changed() {
+		/**
+			 * Indicate that the HTML output of front-end has changed.
+			 *
+			 * If there is any page cache, it should be invalidated when this action is triggered.
+			 */
+			do_action( 'jetpack_boost_page_output_changed' );
```

Comment:
Indentation somehow went bananas.

```suggestion
		 * Indicate that the HTML output of front-end has changed.
		 *
		 * If there is any page cache, it should be invalidated when this action is triggered.
		 */
		do_action( 'jetpack_boost_page_output_changed' );
```